client/src/components/MapElement/MapElement.js

```diff
@@ -11,32 +13,88 @@ MapElement.propTypes = {
 export function MapElement({ className }) {
   const appView = useSelector((state) => state.view.appView)
   const activeTripId = useSelector((state) => state.view.activeTripId)
+  const dispatch = useDispatch()
+
+  const [destination, setDestination] = useState('')
+  const [stagesPerDay, setStagesPerDay] = useState('')
+  const [budget, setBudget] = useState('')
+  const [numberOfDays, setNumberOfDays] = useState('')
+  const [invalidForm, setInvalidForm] = useState(false)
+
+  const resetForm = () => {
+    setDestination('')
+    setStagesPerDay('')
+    setBudget('')
+    setNumberOfDays('')
+    setInvalidForm(false)
+  }
+  const handleCloseNewTripModal = () => {
+    dispatch(setAppView(AppView.TRIP_VIEW))
+    resetForm()
+  }
+  const handleSubmitNewTripForm = () => {
+    if (destination && stagesPerDay && budget && numberOfDays) {
+      // api call with form data
+      console.log(destination, stagesPerDay, budget, numberOfDays)
+      handleCloseNewTripModal()
+    } else {
+      setInvalidForm(true)
+    }
+  }
 
   const newTripForm = useMemo(() => {
     return (
-      <div className='z-[5] flex h-full w-full flex-row items-center justify-center'>
-        <span className='inline-flex w-1/2 flex-col justify-center rounded-md bg-slate-100 bg-opacity-80 p-4'>
-          <h1 className='pb-8 font-mono text-5xl'>Dear Traveler...</h1>
-          <form className='flex w-full flex-col items-center'>
-            <input
-              placeholder='Tell me where you want to go...'
-              className='mb-2 w-2/3 rounded-md border-4 border-indigo-950 p-2'
-            />
-            <input
-              placeholder="Tell me how long you're going for..."
-              className='mb-2 w-2/3 rounded-md border-4 border-indigo-950 p-2'
-            />
-          </form>
+      <Modal
+        open={appView === AppView.NEW_TRIP}
+        handleClose={handleCloseNewTripModal}
+        title='Manifesting A New Trip...'
+        footer={
           <Button
-            onClick={() => {}}
-            className='font-bolder mt-4 bg-slate-300 hover:bg-slate-400'
+            className='font-bolder mt-4 w-full bg-slate-300 hover:bg-slate-400'
+            onClick={handleSubmitNewTripForm}
+            type='submit'
           >
             Brew 🍵
```

Comment:
Is it possible to make this button narrower, to match the inputs? 

![image](https://github.com/cpsc455-bugstorm/TravelersTea/assets/30674558/06a2d93e-bfbd-43ad-896c-249db660b4ff)

Reply:
And if it's not too hard, make the rounding of the modal & button match (if it's not possible bc of MUI then it's ok)